### PR TITLE
feat(domain): create post

### LINKED ISSUE #139: Create post
Now we have an editor, it's time to create our first posts!

### PR DESCRIPTION
**Pull request by petermasking**
Fixes #139

Changes proposed in this pull request:
- improvement of the image module
- implementation of the create post function
- implementation of the first saga testing strategy

@MaskingTechnology/comify

### CODE REVIEW

**Review by basmasking — CHANGES_REQUESTED** [review]

File: src/integrations/validation/definitions/interfaces.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-import { ValidationSchema } from './types';
+import { ValidationSchema } from './types.js';
```
[the remark this answers is not in the document]
Comment (petermasking) [reply src/integrations/validation/definitions/interfaces.ts:2]: I needed to add this in order to make it work. I think this is an issue with Jitar, but I'm not sure yet. I'm going to figure this out later.

File: src/integrations/validation/implementations/zod/Zod.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 
 import { ZodError, ZodIssue, ZodType, z } from 'zod';
-import { FieldTypes, MAX_EMAIL_LENGTH } from '../../definitions/constants';
-import { InvalidData, UnknownValidator } from '../../definitions/errors';
-import { Validator } from '../../definitions/interfaces';
-import { Message, ValidationSchema, Validations } from '../../definitions/types';
+import { FieldTypes, MAX_EMAIL_LENGTH } from '../../definitions/constants.js';
```
[the remark this answers is not in the document]
Comment (petermasking) [reply src/integrations/validation/implementations/zod/Zod.ts:3]: I needed to add this in order to make it work. I think this is an issue with Jitar, but I'm not sure yet. I'm going to figure this out later.

File: src/integrations/validation/implementations/zod/create.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 
-import Zod from './Zod';
+import Zod from './Zod.js';
```
[the remark this answers is not in the document]
Comment (petermasking) [reply src/integrations/validation/implementations/zod/create.ts:2]: I needed to add this in order to make it work. I think this is an issue with Jitar, but I'm not sure yet. I'm going to figure this out later.

File: src/integrations/validation/module.ts [hunk 4]
```diff
@@ -1,2 +1,6 @@
 
-export { default } from './implementation';
+export * from './definitions/constants.js';
```
[the remark this answers is not in the document]
Comment (petermasking) [reply src/integrations/validation/module.ts:2]: I needed to add this in order to make it work. I think this is an issue with Jitar, but I'm not sure yet. I'm going to figure this out later.

**Review by basmasking — APPROVED** [review]

**Pull request merged**